File: exercise/c#/Day16/Day16.Tests/ArticleBuilder.cs
namespace Day16.Tests
{
    public class ArticleBuilder
    {
        public const string Author = "Pablo Escobar";
        public const string CommentText = "Amazing article !!!";
        private readonly Dictionary<string, string> _comments = new();

        public static ArticleBuilder AnArticle() => new();

        public ArticleBuilder Commented()
        {
            _comments.Add(CommentText, Author);
            return this;
        }

        public Article Build()
        {
            var article = new Article(
                "Lorem Ipsum",
                "consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore"
            );

            _comments
                .ToList()
                .ForEach(comment => article.AddComment(comment.Key, comment.Value));

            return article;
        }
    }
}

File: exercise/kotlin/day16/src/test/kotlin/ArticleBuilder.kt
import blog.Article

class ArticleBuilder {
    private val comments: MutableMap<String, String> = mutableMapOf()

    fun commented(): ArticleBuilder = apply { comments += (COMMENT_TEXT to AUTHOR) }

    fun build(): Article {
        val article = Article("Lorem Ipsum", "consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore")
        comments.forEach { (text, author) -> article.addComment(text, author) }
        return article
    }

    companion object {
        const val AUTHOR = "Pablo Escobar"
        const val COMMENT_TEXT = "Amazing article !!!"
        fun anArticle(): ArticleBuilder = ArticleBuilder()
    }
}